mrmustard/physics/wires.py

```diff
@@ -169,242 +310,255 @@ def __init__(
         modes_in_ket: set[int] | None = None,
         classical_out: set[int] | None = None,
         classical_in: set[int] | None = None,
-    ) -> None:
-        self.args: tuple[set, ...] = (
-            modes_out_bra or set(),
-            modes_in_bra or set(),
-            modes_out_ket or set(),
-            modes_in_ket or set(),
-            classical_out or set(),
-            classical_in or set(),
+    ):
+        if any(
+            not (isinstance(g, set) or g is None)
+            for g in (
+                modes_out_bra,
+                modes_in_bra,
+                modes_out_ket,
+                modes_in_ket,
+                classical_out,
+                classical_in,
+            )
+        ):
+            raise ValueError(
+                f"All arguments must be sets or None. got {type(modes_out_bra)}, {type(modes_in_bra)}, {type(modes_out_ket)}, {type(modes_in_ket)}, {type(classical_out)}, {type(classical_in)}"
+            )
+
+        modes_out_bra = modes_out_bra or set()
+        modes_in_bra = modes_in_bra or set()
+        modes_out_ket = modes_out_ket or set()
+        modes_in_ket = modes_in_ket or set()
+        classical_out = classical_out or set()
+        classical_in = classical_in or set()
+
+        self.quantum_wires = set()
+        self.classical_wires = set()
+        for i, m in enumerate(sorted(modes_out_bra)):
+            self.quantum_wires.add(QuantumWire(mode=m, is_out=True, is_ket=False, index=i))
+        n = len(modes_out_bra)
+        for i, m in enumerate(sorted(modes_in_bra)):
+            self.quantum_wires.add(QuantumWire(mode=m, is_out=False, is_ket=False, index=n + i))
+        n += len(modes_in_bra)
+        for i, m in enumerate(sorted(modes_out_ket)):
+            self.quantum_wires.add(QuantumWire(mode=m, is_out=True, is_ket=True, index=n + i))
+        n += len(modes_out_ket)
+        for i, m in enumerate(sorted(modes_in_ket)):
+            self.quantum_wires.add(QuantumWire(mode=m, is_out=False, is_ket=True, index=n + i))
+        n += len(modes_in_ket)
+        for i, m in enumerate(sorted(classical_out)):
+            self.classical_wires.add(ClassicalWire(mode=m, is_out=True, index=n + i))
+        n += len(classical_out)
+        for i, m in enumerate(sorted(classical_in)):
+            self.classical_wires.add(ClassicalWire(mode=m, is_out=False, index=n + i))
+
+    @classmethod
```

Comment:
Is this a classmethod or a staticmethod?  I can't seem to see a reference to 'cls' in the body.  I think it's static, as I can't think of a case where a subclass would need this.  But if it is, then the first line should probably be 'w = cls()'.